### PR TITLE
[library usage] Implement library token provider

### PR DESCRIPTION
**Pull request by bcho**
This pull request is part of the changes for #373

In this pull request, we implemented the library token provider instance and options settings. To allow reusing the environment variables, we introduced a new package `internal/env` for holding all environment variables.

### CODE REVIEW

File: pkg/token/options.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package token
+
+import "github.com/Azure/kubelogin/pkg/internal/token"
+
+// list of supported login methods for library consumers
+
+const (
+	ServicePrincipalLogin = token.ServicePrincipalLogin
+	ROPCLogin             = token.ROPCLogin
```
Comment (weinong) [inline pkg/token/options.go:9]: i think we should drop ropc unless someone requested it

File: pkg/token/types.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// AccessToken represents an Azure service bearer access token with expiry information.
+type AccessToken = azcore.AccessToken
+
+// TokenProvider provides access to tokens.
+type TokenProvider interface {
+	// GetAccessToken returns an access token from giving settings.
```
Comment (weinong) [inline pkg/token/types.go:14]:
```suggestion
	// GetAccessToken returns an access token from given settings.
```

File: pkg/token/options.go (hunk 1 above)
Comment (bcho) [reply pkg/token/options.go]: done, dropped the `ROPCLogin` method

**Review by weinong — APPROVED** [review]
Left a non-blocking comment.

**Pull request merged**